### PR TITLE
Branch income description

### LINKED ISSUE #96: Add cashflow description
As a user, I can add cashflow description so that I can save more detailed information relating to a cashflow

### PR DESCRIPTION
**Pull request by NeoMinWei**
Closes #96

### CODE REVIEW

File: src/main/java/seedu/financialplanner/commands/AddCashflowCommand.java [hunk 1]
```diff
@@ -91,6 +92,12 @@ public AddCashflowCommand(RawCommand rawCommand) throws IllegalArgumentException
             throw new IllegalArgumentException("Recurring value cannot be negative");
         }
 
+        if (rawCommand.extraArgs.containsKey("d")) {
+            logger.log(Level.INFO, "Getting description of cashflow");
+            description = rawCommand.extraArgs.get("d");
+        }
+        rawCommand.extraArgs.remove("d");
+
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/commands/AddCashflowCommand.java]: I like that we can add descriptions now

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 2]
```diff
@@ -95,37 +95,50 @@ private static boolean createNewFile() {
 
         return line.equalsIgnoreCase("y");
     }
-
+    private static boolean hasDescription(String[] split) {
+        return (split.length > 4);
+    }
     private static Cashflow getEntry(String type, String[] split)
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/storage/LoadData.java]: Maybe can add spaces between methods to make the code more readable?

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 3]
```diff
@@ -95,37 +95,50 @@ private static boolean createNewFile() {
...
 
         switch (type) {
         case "I":
             value = Double.parseDouble(split[1].trim());
             recur = Integer.parseInt(split[3].trim());
+            if (hasDescription(split)) {
+                description = split[4].trim();
+            } else {
+                description = null;
+            }
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/storage/LoadData.java]: I think can extract the if-else block in both cases "I" and "E" into the hasDescription method and just make it getDescription?

**Review by ryan1604 — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/cashflow/Cashflow.java [hunk 4]
```diff
@@ -77,7 +81,13 @@ public static double getBalance() {
     }
 
     public String formatString() {
-        return " | " + this.recur;
+        String string = " | " + this.recur;
+
+        if (description != null) {
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/cashflow/Cashflow.java:86]: Nice formatting

File: src/main/java/seedu/financialplanner/cashflow/Cashflow.java [hunk 5]
```diff
@@ -12,10 +12,12 @@ public abstract class Cashflow {
     protected static double balance = 0;
     protected double amount;
     protected int recur;
+    protected String description;
 
-    public Cashflow(double amount, int recur) {
+    public Cashflow(double amount, int recur, String description) {
         this.amount = amount;
         this.recur = recur;
+        this.description = description;
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/cashflow/Cashflow.java:20]: Remember to check for when description is "" exception

File: src/main/java/seedu/financialplanner/cashflow/CashflowList.java [hunk 6]
```diff
@@ -24,31 +24,31 @@ public static CashflowList getInstance() {
...
+        Expense toAdd = new Expense(value, type, recur, description);
         list.add(toAdd);
         ui.printAddedCashflow(toAdd);
 
         int newListSize = list.size();
         assert newListSize == existingListSize + 1;
     }
 
-    public double delete(int index) {
+    public double deleteCashflowWithoutCategory(int index) {
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/cashflow/CashflowList.java:51]: What is meant by cahsflow without category

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 7]
```diff
@@ -96,36 +96,51 @@ private static boolean createNewFile() {
...
             } catch (IllegalArgumentException e) {
                 throw new IllegalArgumentException("Invalid income type");
             }
-            entry = new Income(value, incomeType, recur);
+            entry = new Income(value, incomeType, recur, description);
             break;
         case "E":
             value = Double.parseDouble(split[1].trim());
             recur = Integer.parseInt(split[3].trim());
+            if (hasDescription(split)) {
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/storage/LoadData.java]: Looks a little messy. Might want to use java built in serialize

File: src/test/java/seedu/financialplanner/cashflow/CashflowListTest.java [hunk 8]
```diff
@@ -28,8 +28,9 @@ void testAddIncomeAndExpense() {
         assertEquals(IncomeType.SALARY, testIncome.getIncomeType());
         assertEquals(30, testIncome.recur);
         assertEquals("15.00", decimalFormat.format(roundedBalance));
+        assertEquals("part time job", testIncome.description);
 
-        testList.addIncome(15.999, IncomeType.INVESTMENTS, 0);
+        testList.addIncome(15.999, IncomeType.INVESTMENTS, 0, "AAPL");
```
Comment (wwweert123) [inline src/test/java/seedu/financialplanner/cashflow/CashflowListTest.java:33]: Wow. Adding investments as well!

**Review by wwweert123 — APPROVED** [review]
Looks good

**Pull request merged**